game/src/js/objects/ui/DOMModal.js

```diff
@@ -56,25 +57,34 @@ export default class DOMModal {
       });
     });
     // Process data into HTML
+    
     if(this.props.data) {
       const objResolve = function(rootObj, string) {
         let value = rootObj;
         let parts = string.split('.');
         while(parts.length != 0) {
           value = value[parts.shift()];
         }
+        
         return value;
       }
-      this.modal.querySelectorAll('*[data-value]').forEach(element => {
-        if(element.nodeName == 'INPUT') {
+      
+      this.modal.querySelectorAll('*[data-checked]').forEach(element => {
+        if(element.nodeName == 'INPUT' && element.type != 'checkbox') {
```

Comment:
I'd put the `element.type == 'checkbox'` first, so that the `else if` below can just check for `element.nodeName == 'INPUT'`.

```javascript
if(element.nodeName == 'INPUT' && element.type == 'checkbox') {
  ...
}
else if(element.nodeName == 'INPUT') {
  ...
```